frontend/src/metabase/home/containers/SearchApp.jsx

```diff
@@ -39,7 +49,12 @@ export default class SearchApp extends React.Component {
                   </Flex>
                 );
               }
-              const types = _.chain(list)
+
+              const types = _.chain(
+                location.query.type
+                  ? list.filter(l => l.model === location.query.type)
```

Comment:
Ideally `model` would just be a parameter on the search endpoint, but this is ok for now.